src/Bicep.Core/TypeSystem/DeployTimeConstantValidator.cs

```diff
@@ -8,31 +8,107 @@
 using Bicep.Core.Extensions;
 using Bicep.Core.Semantics;
 using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem.Az;
 
 namespace Bicep.Core.TypeSystem
 {
     public static class DeployTimeConstantValidator
     {
         public static void Validate(SemanticModel semanticModel, IDiagnosticWriter diagnosticWriter)
         {
-            // Collect all sytnaxes that require DTCs (a.k.a. DTC containers).
-            var containers = DeployTimeConstantContainerVisitor.CollectDeployTimeConstantContainers(semanticModel);
+            var existingResourceBodyObjectTypeOverrides = new Dictionary<DeclaredSymbol, ObjectType>();
+
+            // grab all existing resources
+            var existingResourceSymbols = semanticModel.DeclaredResources
+                .Select(x => x.Symbol)
+                .Where(x => x.DeclaringResource.IsExistingResource())
+                .ToArray();
+
+            // compare existing resources to all other existing resources to find dependencies and see if it has a non-DTC value
+            // and needs to remove the ReadableAtDeployTime flag from the "name" property -> O(n^2) time complexity
+            for (int i = 0; i < existingResourceSymbols.Length; i++)
+            {
+                foreach (var existingResourceSymbol in existingResourceSymbols)
+                {
```

Comment:
There are two optimizations we could do:
1. At the beginning of the inner loop, check if `existingResourceSymbol` already exists in `existingResourceBodyObjectTypeOverrides`. If yes, continue the inner loop to skip the current iteration.
2. If no new item is added to `existingResourceBodyObjectTypeOverrides` at the end of the outer loop, break the outer loop.

Reply:
Great point!